app/models/host.rb

```diff
@@ -1934,19 +1934,18 @@ def has_compliance_policies?
   end
 
   def self.node_types
-    return :mixed_hosts if count_of_openstack_hosts > 0 && count_of_non_openstack_hosts > 0
-    return :openstack   if count_of_openstack_hosts > 0
-    :non_openstack
+    return :non_openstack unless openstack_hosts_exists?
+    non_openstack_hosts_exists? ? :mixed_hosts : :openstack
   end
 
-  def self.count_of_openstack_hosts
+  def self.openstack_hosts_exists?
     ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where(:ems_id => ems).count
+    Host.where(:ems_id => ems).exists? unless ems.empty?
```

Comment:
why not the clearer:
```ruby
return false if ems.empty?
Host.where(:ems_id => ems).exists?
```

I always have a hard time groking boolean methods that do not end in boolean methods.